pxtlib/service.ts

```diff
@@ -384,23 +385,31 @@ namespace ts.pxtc {
         const combinedGet: pxt.Map<SymbolInfo> = {}
         const combinedChange: pxt.Map<SymbolInfo> = {}
 
-        function addCombined(tp: string, s: SymbolInfo) {
-            const isGet = tp == "get"
-            const m = isGet ? combinedGet :
-                tp == "set" ? combinedSet : combinedChange
-            let ex = U.lookup(m, s.namespace)
+        function addCombined(rtp: string, s: SymbolInfo) {
+            const isGet = rtp == "get"
+            const isSet = rtp == "set"
+            const m = isGet ? combinedGet : (isSet ? combinedSet : combinedChange)
+            const mkey = `${s.namespace}.${s.retType}`
+
+            let ex = U.lookup(m, mkey)
             if (!ex) {
-                let paramName = s.namespace.toLowerCase()
-                ex = m[s.namespace] = {
+                const tp = `@${rtp}@`
+                const paramName = s.namespace.toLowerCase()
+                const paramValue = `value=${s.attributes.blockCombineShadow || ""}`;
+
+                ex = m[mkey] = {
                     attributes: {
+                        blockId: `${mkey}_blockCombine_${rtp}`,
```

Comment:
Changing the blockId is a breaking change for existing scripts. It's probably fine in this case since we don't have any editors that use this feature right now.

Reply:
If you wanted to avoid the break, you could special case `number` (but it probably doesn't matter)

Reply:
Only used in Arcade right now

Reply:
Good point. I'll special case generating the blockid re: number type. I'll update the PR.

Reply:
Done. PR updated.

Validated with project scripts that pre-existed this change and that were generated with the old format blockId for number property types.